detox/src/logger/utils/streamUtils.js

```diff
@@ -0,0 +1,240 @@
+}
+
+function mergeSortedJSONL(jsonlStreams, comparator = compareTimestamps) {
+  const intermediate = mapTransform(extractValue);
+  const reemitError = (err) => intermediate.emit('error', err);
+  const addReemit = (stream) => stream.on('error', reemitError);
+  return pipe(multiSort(jsonlStreams.map(addReemit), comparator), intermediate, through());
+}
+
+function writeJSONL() {
+  return new JsonlStringer({
+    header: '',
+    delimiter: '\n',
+    footer: '',
+  });
+}
+
+function writeJSON() {
+  return new JsonlStringer({
+    header: '[\n\t',
+    delimiter: ',\n\t',
+    footer: '\n]\n',
+  });
+}
+
+function debugStream(options) {
+  const out = new PassThrough({ encoding: 'utf8' });
+  const writable = bunyanDebugStream.default({
+    ...options,
+    colors: false,
+    out,
+  });
+
+  return duplexify(writable, out, {
+    readableObjectMode: false,
+    writableObjectMode: true
+  });
+}
+
+class JsonlStringer extends Transform {
+  constructor({ replacer = undefined, header = '', delimiter = '\n', footer = '' } = {}) {
+    super({ writableObjectMode: true, readableObjectMode: false });
+
+    this._replacer = replacer;
+    this._header = header;
+    this._delimiter = delimiter;
+    this._footer = footer;
+  }
+
+  _transform(chunk, _, callback) {
+    if (this._header) {
+      this.push(this._header);
+    }
+
+    this.push(JSON.stringify(chunk, this._replacer));
+    this._transform = this._nextTransform;
+    callback(null);
+  }
+
+  _nextTransform(chunk, _, callback) {
+    this.push(this._delimiter + JSON.stringify(chunk, this._replacer));
+    callback(null);
+  }
+
+  _flush(callback) {
+    if (this._footer) {
+      this.push(this._footer);
+    }
+
+    callback();
+  }
+}
+
+class SimpleEventBuilder extends AbstractEventBuilder {
+  events = [];
+
+  send(event) {
+    this.events.push(event);
+  }
+}
+
+function chromeTraceStream() {
+  const knownPids = new Set();
+  const knownTids = new Set();
+
+  return flatMapTransform((data) => {
+    // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+    const { cat = 'default', msg: name, ph = 'i', pid, tid, time, name: _name, hostname: _hostname, ...args } = data;
+    const ts = new Date(time).getTime() * 1E3;
+
+    const builder = new SimpleEventBuilder();
+    if (!knownPids.has(pid)) {
+      builder.process_name(pid === process.pid ? 'primary' : 'secondary', pid);
+      knownPids.add(pid);
+    }
+
+    const tidHash = `${pid}:${tid}`;
+    if (!knownTids.has(tidHash)) {
+      const primaryCategory = cat.split(',', 1)[0];
+      builder.thread_name(primaryCategory, tid, pid);
+      builder.thread_sort_index(tid, tid, pid);
+      knownTids.add(tidHash);
+    }
+
+    const event = { ph, name, pid, tid, cat, ts, args };
+    if (ph === 'E') {
+      delete event.name;
+    }
+
+    builder.events.push(event);
+    return builder.events;
+  });
+}
+
+// TODO: create PR to https://github.com/mafintosh/duplexify
```

Comment:
Just more info on this. I tried to create a PR, but my skills with streams were insufficient to produce a correct unit test for what I wanted. 😭